src/core/AutoRest/Simplify/AdHocEqualityComparer.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// 
+
+using System;
+using System.Collections.Generic;
+
+namespace AutoRest.Simplify
+{
+    public class AdHocEqualityComparer<T> : IEqualityComparer<T>
```

Comment:
Think we added one of those today :D 
see `AutoRest.Core.Utilities.EqualityComparer`